### ISSUE TITLE
How to apply transfer learning /continous Learning with small annotated dataset ?

### ISSUE THREAD

**Issue by TapendraBaduwal** [ultralytics/ultralytics#2338]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Lets we have 80 class label..  i want to finetune a existing model with new labels or old labels mix.

In new annotation data there may be the new id class for same label name as existing model have...how model handel this?

My new train model must detect existing label and new labels... how we can do that ?

If we annotated new label then shall we need to increase manually number of classes and add label name manually?

If existing model label name is "person" and my new annotated label name is "Person" where P capital then after training model learn same or not?

**Comment by glenn-jocher** [comment]
@TapendraBaduwal hello,

If you have added new labels to your annotation data, you will need to increase the number of classes manually in your existing model and add the new label names manually. Additionally, if you have added a new label with a similar name to an existing label, the model may not be able to distinguish between them during training because it only sees the numerical class IDs. We recommend using unique label names to avoid this issue. 

After training, the model may be able to detect the new and existing labels, but this also depends on the quality and quantity of the new data that you are using to fine-tune the model.

I hope this helps! Let me know if you have any other questions.

**Comment by TapendraBaduwal** [comment]
@glenn-jocher i am confused  Lets i have coco128.yaml file in below way..

# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: ../datasets/coco128  # dataset root dir
train: images/train2017  # train images (relative to 'path') 128 images
val: images/train2017  # val images (relative to 'path') 128 images
test:  # test images (optional)

# Classes
names:
  0: person
  1: bicycle
  2: car
  3: motorcycle
  4: airplane
  5: bus
.........

But after annotated i got this one file
train: ../train/images
val: ../valid/images
test: ../test/images

nc: 3
names: ['bus', 'person', 'laptop']

roboflow:
  workspace: pbs
  project: vision-i9skj
  version: 1
  license: MIT
  url: https://universe.roboflow.com/pbs/vision-i9skj/dataset/1...

I am going to fine tune existing model  while fine tune i have to give path as follow 
results = model.train(data='newdata.yaml', epochs=3)  # train the model

Here the new annotation  bus and person  label same as existing and lets "laptop" is new label 

Now in which file we have to add new label "laptop" because i am doing finetune in new data.yaml ..we are not using coco128.yaml file while finetune?

**Comment by glenn-jocher** [comment]
@TapendraBaduwal hello,

If you want to fine-tune an existing model with a new set of labels, then you need to create a new YAML file for the new data that includes all the new labels. In this case, you would create a new YAML file named "newdata.yaml" and add the "laptop" label to it in addition to the existing "bus" and "person" labels. 

When you fine-tune the model using this new data file, the model will be trained to recognize all three labels ("bus", "person", and "laptop") as distinct objects. The model maps the label names to numerical class IDs internally, so you do not need to worry about label naming conventions as long as they are consistent across the data and the YAML file.

I hope that answers your question. Let me know if you have any further questions.
